src/SnowflakeTransformation.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\SnowflakeTransformation;
+
+use Keboola\Component\UserException;
+use Keboola\Datatype\Definition\Exception\InvalidTypeException;
+use Keboola\Datatype\Definition\GenericStorage as GenericDatatype;
+use Keboola\Datatype\Definition\Snowflake as SnowflakeDatatype;
+use Keboola\SnowflakeDbAdapter\Connection;
+use Keboola\SnowflakeDbAdapter\QueryBuilder;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Serializer\Encoder\JsonEncode;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+
+class SnowflakeTransformation
+{
+    private Connection $connection;
+
+    private LoggerInterface $logger;
+
+    private Config $config;
+
+    private array $databaseConfig;
+
+    public function __construct(Config $config, LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+        $this->config = $config;
+        $this->databaseConfig = $config->getDatabaseConfig();
+        $this->connection = $this->createConnection();
+    }
+
+    public function createManifestMetadata(array $tables, string $dataDir): void
+    {
+        $getTables = $this->getTables($tables);
+        foreach ($getTables as $getTable) {
+            $tableName = $getTable['name'];
+            $outputMappingTable = array_filter($tables, function ($item) use ($tableName) {
+                if ($item['source'] !== $tableName) {
+                    return false;
+                }
+                return true;
+            });
+            $outputMappingTable = array_values($outputMappingTable);
+            $manifestData = [
+                'destination' => $outputMappingTable[0]['destination'],
+                'column_metadata' => [],
+            ];
+
+            foreach ($getTable['columns'] as $column) {
+                $datatypeKeys = ['length', 'nullable', 'default'];
+                try {
+                    $datatype = new SnowflakeDatatype(
+                        $column['type'],
+                        array_intersect_key($column, array_flip($datatypeKeys))
+                    );
+                } catch (InvalidTypeException $e) {
+                    $datatype = new GenericDatatype(
+                        $column['type'],
+                        array_intersect_key($column, array_flip($datatypeKeys))
+                    );
+                }
+                $columnMetadata = $datatype->toMetadata();
+                $nonDatatypeKeys = array_diff_key($column, array_flip($datatypeKeys));
+                foreach ($nonDatatypeKeys as $key => $value) {
+                    if ($key !== 'name') {
+                        $columnMetadata[] = [
+                            'key' => 'KBC.' . $key,
+                            'value'=> $value,
+                        ];
+                    }
+                }
+                $manifestData['column_metadata'][$column['name']] = $columnMetadata;
```

Comment:
a to nejdulezitejsi $manifestData['columns']

Reply:
nestačí vyplněný `column_metadata` https://github.com/keboola/php-component/blob/master/src/Manifest/ManifestManager/Options/OutTableManifestOptions.php#L29 ?

Reply:
nn, to je neco uplne jinyho https://developers.keboola.com/extend/common-interface/manifest-files/#dataouttables-manifests

Reply:
ok, přidal jsem to